### DISCUSSION TITLE
How to avoid redirect to localhost or 127.0.0.1

### DISCUSSION THREAD

**Discussion by discoverlance-com** [php/frankenphp#759] [Q&A]
I am quite new to using octane and particularly frankenphp and I am running into issues with the redirection of the host.

For now, I am using laravel sail to run my application so after I installed laravel octane and used the frankenphp server, I updated my docker-compose environment to now be as follows per the laravel docs:

```docker-compose
environment:
            WWWUSER: '${WWWUSER}'
            LARAVEL_SAIL: 1
            XDEBUG_MODE: '${SAIL_XDEBUG_MODE:-off}'
            XDEBUG_CONFIG: '${SAIL_XDEBUG_CONFIG:-client_host=host.docker.internal}'
            IGNITION_LOCAL_SITES_PATH: '${PWD}'
            SUPERVISOR_PHP_COMMAND: '/usr/bin/php -d variables_order=EGPCS /var/www/html/artisan octane:start --server=frankenphp --host=localhost --admin-port=2019 --port=80'
            XDG_CONFIG_HOME: /var/www/html/config
            XDG_DATA_HOME: /var/www/html/data
```

I also have the following in my environment variable:

```
APP_URL=https://example-domain.com
ASSET_URL="${APP_URL}"
APP_PORT=8000
```

This works alright and I am able to load my application, but for some reason, anytime I try to click a link on my application, I get redirected to localhost, as an example when I click on the about page link, this is where I get redirected to: `https://127.0.0.1:8000/about` which is invalid. I am not sure how to set the url so it uses my application's url.

I experience a similar thing when I try to deploy the application with docker. I have the following docker file but on that page. But after building the docker image and trying to run the image with the command: `docker run -p 8000:80 --env-file .env`, I have similar issues, thus, all links have the 127.0.0.1 as the url(https://127.0.0.1:8000/about) and another weird issue in my generated html is that I get this picture below as the urls generated for my vite assets:

![image](https://github.com/dunglas/frankenphp/assets/52635726/cb774c08-f8e3-43cf-a583-5fd6fb265434)

```docker
FROM dunglas/frankenphp

ARG NODE_VERSION=20

RUN apt-get update && apt-get install -y libssl-dev zlib1g-dev \
    libzip-dev \
    libonig-dev \
    libsodium-dev \
    libpng-dev \
    libjpeg62-turbo-dev \
    libfreetype6-dev \
    libc-client-dev libkrb5-dev \
    zip \
    unzip \
    curl \
    unzip \
    gnupg \ 
    gosu \ 
    ca-certificates \ 
    sqlite3 \ 
    libcap2-bin \
    libpng-dev \
    librsvg2-bin \
    build-essential \
    ffmpeg

RUN install-php-extensions pcntl \
    mysqli \
    bcmath \
    sodium \
    exif \
    gd \ 
    imap \
    intl \
    pdo_mysql \
    zip \
    mbstring 

COPY . /app

RUN curl -sS https://getcomposer.org/installer -o composer-setup.php

RUN php composer-setup.php --install-dir=/usr/local/bin --filename=composer

RUN composer install --no-cache --no-dev --no-scripts --no-autoloader --ansi --no-interaction
RUN composer dump-autoload -o

RUN curl -fsSL https://deb.nodesource.com/setup_current.x | bash - && \
  apt-get install -y nodejs \
  build-essential && \
  node --version && \ 
  npm --version && \
  npm install && \
  npm run build

ENTRYPOINT ["php", "artisan", "octane:frankenphp", "--port=80", "--admin-port=2019"]

```

I am using a workstation like gitpod and it basically generates an https url so I don't need to use the automatic ssl generation that comes with frankenphp/caddy.

**Comment by withinboredom** [comment]
I recommend taking a look at https://github.com/dunglas/frankenphp/discussions/718 which might help.
